_cffi_build/dice_one_against_many.cpp

```diff
@@ -315,9 +377,11 @@ extern "C"
             const char *arrays, int narrays, int array_bytes) {
         // assumes WORD_BYTES divides array_bytes
         int nwords = array_bytes / WORD_BYTES;
-        const uint64_t *u = reinterpret_cast<const uint64_t *>(arrays);
+        // The static_cast is to avoid int overflow in the multiplication
+        size_t total_bytes = static_cast<size_t>(narrays) * array_bytes;
+        auto ptr = adjust_ptr_alignment(arrays, total_bytes);
+        auto u = ptr.get();
```

Comment:
Is u really such a complicated type it needs to be auto? (It just feels less understandable to my old school C++ prejudices).